lib/pagseguro/collection_base.rb

```diff
@@ -0,0 +1,41 @@
+module PagSeguro
+  class CollectionBase
+    extend CollectionManagers::MacroMethods
+    extend Forwardable
+    include Enumerable
+
+    def_delegators :@store, :size, :clear, :empty?, :any?, :each
+
+    def initialize
+      @store = []
+    end
+
+    def collection_manager
+      CollectionManagers::GenericManager
+    end
+
+    def collection_type
+      eval singularize(self.class.to_s)
```

Comment:
ah ok, but ruby 1.9.3 and 2.0.0 has const_get 
http://ruby-doc.org/core-1.9.3/Module.html#method-i-const_get
http://ruby-doc.org/core-2.0.0/Module.html#method-i-const_get